scrapy/core/engine.py

```diff
@@ -311,30 +293,20 @@ def _schedule_request(self, request: Request, spider: Spider) -> None:
                 signals.request_dropped, request=request, spider=spider
             )
 
-    def download(self, request: Request, spider: Optional[Spider] = None) -> Deferred:
+    def download(self, request: Request) -> Deferred:
         """Return a Deferred which fires with a Response as result, only downloader middlewares are applied"""
-        if spider is not None:
-            warnings.warn(
-                "Passing a 'spider' argument to ExecutionEngine.download is deprecated",
-                category=ScrapyDeprecationWarning,
-                stacklevel=2,
-            )
-            if spider is not self.spider:
-                logger.warning(
-                    "The spider '%s' does not match the open spider", spider.name
-                )
         if self.spider is None:
             raise RuntimeError(f"No open spider to crawl: {request}")
-        return self._download(request, spider).addBoth(
-            self._downloaded, request, spider
+        return self._download(request, self.spider).addBoth(
+            self._downloaded, request, self.spider
```

Comment:
Maybe remove this parameter altogether from `_download` function? Since we have this logic inside the function?

https://github.com/scrapy/scrapy/blob/9f9a2292e08cb0944e1e88e64cef1f4b17486ec8/scrapy/core/engine.py#L344-L345

Reply:
What you think? @wRAR